models/serial.js

```diff
@@ -0,0 +1,32 @@
+var mongoose = require('mongoose')
+var Schema = mongoose.Schema
+var when = require('when')
+
+var START = 100000 // leave room for old serials
+
+var serialSchema = new mongoose.Schema({
+  _id: { type: String },
+  counter: { type: Number, default: Date.now() }
+})
+
+serialSchema.statics.next = function(name) {
+	var dfd = when.defer()
+
+	this.findOneAndUpdate(
+		{ _id: name },
+		{ $inc: { counter: 1 } },
+		{ upsert: true, new: true },
+		function(err, result) {
+			if (err) {
+				return dfd.reject(err)
+			}
+
+			return dfd.resolve(START + result.counter)
```

Comment:
if we ever change START, this will introduce offsets retroactively?

Reply:
I don't understand. We can't really change START after this. This is there so that existing edit logs don't break (because edits would start from 0). Not the best thing, we could do without START, but then some edit log sessions could potentially break.